contributing.md

```diff
@@ -0,0 +1,105 @@
+# guidelines for contributing PR's, code style guide, guidelines for issues, forum
+This document should grow over time. These are just guidelines not hard rules (depending on circumstances).
+Rationale should be provided for contentious guidelines.
+
+## git
+### use `git rebase` instead of `git merge`
+In command line (for person doing PR):
```

Comment:
I don't care one way or the other, but is this currently one of the guidelines used when contributing? I feel this PR should document the guidelines and best practices that are in place, not invent new ones

Reply:
To be more clear: I think that imposing all this git trickery will only discourage useful PRs. I think I have seen many PRs that were not rebased or squashed, but were accepted anyway